src/main/java/org/candlepin/async/tasks/CertificateCleanupJob.java

```diff
@@ -0,0 +1,96 @@
+import org.candlepin.model.IdentityCertificate;
+import org.candlepin.model.IdentityCertificateCurator;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * The job will periodically delete all expired identity and content certificates
+ * as well as all revoked and expired certificate serials.
+ */
+public class CertificateCleanupJob implements AsyncJob {
+    private static final Logger log = LoggerFactory.getLogger(CertificateCleanupJob.class);
+
+    public static final String JOB_KEY = "CertificateCleanupJob";
+    public static final String JOB_NAME = "Certificate Cleanup";
+    public static final String DEFAULT_SCHEDULE = "0 0 12 * * ?";
+
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identCertCurator;
+    private final ContentAccessCertificateCurator caCertCurator;
+    private final CertificateSerialCurator serialCurator;
+
+    @Inject
+    public CertificateCleanupJob(
+        ConsumerCurator consumers,
+        IdentityCertificateCurator identCerts,
+        ContentAccessCertificateCurator contentAccessCerts,
+        CertificateSerialCurator serialCurator) {
+        this.consumerCurator = Objects.requireNonNull(consumers);
+        this.identCertCurator = Objects.requireNonNull(identCerts);
+        this.caCertCurator = Objects.requireNonNull(contentAccessCerts);
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+    }
+
+    @Override
+    @Transactional
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        log.debug("Starting certificate cleanup.");
+        // Skipping entitlement certificates as those are handled by ExpiredPoolsCleanupJob
+        cleanupExpiredIdentityCerts();
+        cleanupExpiredContentAccessCerts();
+        this.serialCurator.deleteRevokedExpiredSerials();
+        log.debug("Certificate cleanup successfully completed.");
+    }
+
+    private void cleanupExpiredIdentityCerts() {
+        List<Consumer> consumers = this.consumerCurator.findAllWithExpiredIdCerts();
+        log.info("Cleaning up {} expired identity certificates.", consumers.size());
+        for (Consumer consumer : consumers) {
+            IdentityCertificate idCert = consumer.getIdCert();
+            consumer.setIdCert(null);
+            this.identCertCurator.delete(idCert);
+        }
+    }
+
+    private void cleanupExpiredContentAccessCerts() {
+        List<Consumer> consumers = this.consumerCurator.findAllWithExpiredContentAccessCerts();
+        log.info("Cleaning up {} expired content access certificates.", consumers.size());
+        for (Consumer consumer : consumers) {
+            ContentAccessCertificate cert = consumer.getContentAccessCert();
+            consumer.setContentAccessCert(null);
+            this.caCertCurator.delete(cert);
```

Comment:
Same thing as for identity certs here: Don't instantiate consumers into memory, use queries, partition queries in batches

Reply:
Done.